### PR TITLE
TRABootCamp-2:Splitting text into chunks

### PR DESCRIPTION
**Pull request by BaraahAlKalbani**
What does this PR do:
-added modules to read a text file and chunk it
-added a module to read user input AND return the response 

User Story:
https://trello.com/c/0DiKBam0/2-splitting-text-into-chunks

Risk (1 Low - 5 High):
1

Impact (1 Low - 5 High):
2

### CODE REVIEW

File: utills/show_user_input.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import streamlit as st
+from openai import OpenAI, load_qa_chain, get_openai_callback
+
+
+# Define a function to display user input and generate responses based on it
+def show_user_input(knowledge_base):
```
Comment (CodelineAtyab) [inline utills/show_user_input.py]: We need to only have methods that can be imported and used by the dev who is going to use streamlit to build a UI

File: utills/show_user_input.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    # Display a text input box for the user to enter a question
+    user_question = st.text_input("Ask a question about your PDF:")
+
+    # Check if the user has entered a question
+    if user_question:
+        # Use the similarity_search function to find documents related to the user's question
+        docs = knowledge_base.similarity_search(user_question)
+
+        # Create an OpenAI object and load a pre-trained QA chain
+        llm = OpenAI()
```
Comment (CodelineAtyab) [inline utills/show_user_input.py]: Violates SRP

**Review by CodelineAtyab — CHANGES_REQUESTED** [review]
Please update as discussed

File: utills/show_user_input.py (hunk 1 above)
Comment (AmeerAllawati) [reply utills/show_user_input.py]: Assume that you are receiving a pdf and a question

**Review by AmeerAllawati — APPROVED** [review]

File: utills/show_user_input.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        Retrieves related documents from a knowledge base based on a user question.
+
+        Args:
+            knowledge_base: The knowledge base to search for related documents.
+            user_question: The user's question.
+
+        Returns:
+            A list of related documents.
+    """
+    if user_question:
```
Comment (CodelineAtyab) [inline utills/show_user_input.py]: In structured programming we have one way in and one way out. It makes the code easy to debug if we have just one return statement.

**Review by CodelineAtyab — APPROVED** [review]
We have to create a ticket to replace print statements with log statements using the logging framework.

**Pull request merged**